src/mesh.cpp

```diff
@@ -235,6 +259,28 @@ bool loadPlyFile(const QString& fileName,
     }
     if (info.colors.size() != info.verts.size())
         info.colors.clear();
+
+    // Reconstruct inner polygons
+    while (innerPolygonVertexCount.size())
+    {
+        unsigned int verticesCount = innerPolygonVertexCount.back();
+        unsigned long firstIdx = info.edges.size();
+        for (size_t j = 0; j < verticesCount; ++j)
+        {
+            if (j > 1)
+                info.edges.push_back(info.edges.back());
+            info.edges.push_back(innerVertexIndices.back());
+            innerVertexIndices.pop_back();
```

Comment:
Agreed on using `while(!innerPolygonVertexCount.empty()))`.

I chose to do it in reverse because `innerPolygonVertexCount` and `innerVertexIndices` can potentially contain information about multiple holes or holes belonging to multiple polygons defined in the ply file.

Because of this I thought it was safer to pop entries off the "stacks" in the opposite order they were inserted. It seemed sligthly less error prone but I agree a little harder to understand.

I can add a comment about the logic or figure out the logic to do it forward.